src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/SocksHelper.cs

```diff
@@ -280,19 +267,13 @@ private static async ValueTask EstablishSocks4TunnelAsync(Stream stream, bool is
                 {
                     ipv4Address.TryWriteBytes(buffer.AsSpan(4), out int bytesWritten);
                     Debug.Assert(bytesWritten == 4);
-                    if (buffer[4] == 0 && buffer[5] == 0 && buffer[6] == 0)
-                    {
-                        // Invalid IP address used by SOCKS4a to represent remote DNS.
-                        // In case we don't have a domain name, throwing.
-                        throw new SocksException(SR.net_socks_ipv4_invalid);
-                    }
```

Comment:
If the client requests `http://0.0.0.1` and the server supports socks4a, it will wait forever for the domain name until timeout. Is there anything prevents `0.0.0.x` passed here? Both `new Uri` and `IPAddress.Parse` allows `0.0.0.x`.

Reply:
@MihaZupan Any idea about this?

Reply:
I don't see this as a concern. The 0.0.0.0/8 blocks sees little actual use due to its historical status under which it wouldn't even be valid as a destination address. This would also only be relevant for socks4a proxies which are less common.